### PR TITLE
Home Assistant docs

### LINKED ISSUE #496: Add Home Assistant-specific documentation
Helping people who want to contribute add:
- [x] Home Assistant backends for existing types (security camera, smoke alarm, thermostat, car)
- [ ] new functionality to existing types
- [x] new types

Should include:
- [x] how to develop
- [x] how to contribute/upload/publish
- [x] how to test locally

### LINKED ISSUE #497: Add documentation around testing
More extensive documentation around:
- [x] unit testing
- [x] integration testing on Web Almond
- [x] testing and training natural language
- [x] testing locally (setting up developer key, updating package, hacking in device class)

### PR DESCRIPTION
**Pull request by gcampax**
Fixes #497 
Fixes #496

### CODE REVIEW

**Comment by gcampax** [comment]
Depends on https://github.com/stanford-oval/almond-cmdline/pull/2 and also releases of Almond all around.

File: doc/home-assistant-integration.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+If you want to have Home Assistant support for an interface that is already in Thingpedia
+for a different brand (for example, the `@security-camera` interface), you should create
+a new module in the `io.home-assistant` folder.
+
+The module should export a class that inherits from the base Home Assistant device defined
+in `./base.js` (and **not** the `Tp.BaseDevice` that Thingpedia devices normally inherit from).
+Inside the class, you can query the Home Assistant entity state as `this.state.state`, and
+query the Home Assistant entity attributes as `this.state.attributes`. To implement actions,
+you can call Home Assistant services with `this._callService`. Please look at `light.js` for
+an example.
```
Comment (rayslxu) [inline doc/home-assistant-integration.md]: Include the code snippet from `light.js` here?

File: doc/home-assistant-integration.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+Add natural language support for the new interface as customary from the `dataset.tt` file.
+Then, add Home Assistant support for the new interface following the previous guide.
+
+By convention, general device interfaces use either a single-part name (`@light-bulb`, `@security-camera`),
+or a name in the `@org.thingpedia` namespace. Only administrators can upload a device with this name though. 
+You should use your own username as prefix (as it is your private namespace), and the name will
+be adjusted during review.
+
```
Comment (rayslxu) [inline doc/home-assistant-integration.md]: "An example of lightbulb can be found here: https://almond.stanford.edu/thingpedia/classes/by-id/light-bulb"

File: doc/thingpedia-cli-tools.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+The command is called `thingpedia` and should be in your path.
+
+## Setup
+
+To create a new project to develop Thingpedia devices, you can run:
+
+```
+thingpedia --developer-key <...> --access-token <...> \
+  init-project my-awesome-devices
```
[the remark this answers is not in the document]
Comment (gcampax) [reply doc/thingpedia-cli-tools.md]: How did you install thingpedia-cli ? It should install lice-js as well.

File: doc/thingpedia-cli-tools.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+## Testing
+
+Tests are specified in `tests/<device-id>.js`. See
+[thingpedia-common-devices](https://github.com/stanford-oval/thingpedia-common-devices)
+for examples on how to write tests.
+
+Once you have tests, you can run them with
+```
+yarn test
```
[the remark this answers is not in the document]
Comment (gcampax) [reply doc/thingpedia-cli-tools.md]: ??? maybe the skeleton is referring to a version of thingpedia-api that is too old?

File: doc/home-assistant-integration.md (hunk 2 above)
Comment (gcampax) [reply doc/home-assistant-integration.md]: Yeah I didn't add links cause I worried the linkchecker would barf. I guess absolute links are fine.

File: doc/thingpedia-cli-tools.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rayslxu) [reply doc/thingpedia-cli-tools.md]: hmm, I was using my local version. yeah, install via yarn solves the problem

File: doc/thingpedia-cli-tools.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gcampax) [reply doc/thingpedia-cli-tools.md]: I updated the skeleton (which btw is the skeleton branch in thingpedia-common-devices), so it should work now.

**Pull request merged**